shell/renderer/electron_render_frame_observer.cc

```diff
@@ -58,20 +58,61 @@ ElectronRenderFrameObserver::ElectronRenderFrameObserver(
 }
 
 void ElectronRenderFrameObserver::DidClearWindowObject() {
+  // Do a delayed Node.js initialization for child window.
+  // Check DidInstallConditionalFeatures below for the background.
+  auto* web_frame = render_frame_->GetWebFrame();
+  if (has_delayed_node_initialization_ && web_frame->Opener() &&
```

Comment:
When browser side navigation is triggered, won't there be another call to `ElectronRenderFrameObserver::DidInstallConditionalFeatures` from https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/bindings/core/v8/local_window_proxy.cc;l=216-220 which will then initialize based on the preferences, is this second redirection via `ElectronRenderFrameObserver::DidClearWindowObject` needed ?

Reply:
For child window opened by `window.open`, the `DidInstallConditionalFeatures` is only called once, and `DidClearWindowObject` is instead called multiple times.

It seems that Chromium immediately creates an empty DOM window when `window.open` is called, and then does an in-place initialization later when the actual navigation happens, it does not create a new DOM window for the navigation.

I think this behavior is kind of reasonable, since the window object returned by `window.open` is immediately accessible to users even before the actual navigation starts.